SSH and TCP definition for Targets (#3513)
hashicorp/boundary: website/content/docs/concepts/domain-model/targets.mdx

- `description` - (optional)

- `type` - (required)

>   Defines the type of connection to a target. Options are `static`, `tcp` and `ssh`. Even if the type of connection to the targer is `ssh`, the `tcp` subcommand should be selected. Using the `ssh` command is reserved for [Credential injection](https://developer.hashicorp.com/boundary/tutorials/hcp-administration/hcp-ssh-cred-injection)

Reviewer: ```suggestion
  Defines the type of connection to a target. Options are `static`, `tcp`, and `ssh`. Even if the type of connection to the target is `ssh`, you should select the `tcp` subcommand. Using the `ssh` command is reserved for [credential injection](https://developer.hashicorp.com/boundary/tutorials/hcp-administration/hcp-ssh-cred-injection)
```

>   Defines the type of connection to a target. Options are `static`, `tcp`, and `ssh`. Even if the type of connection to the target is `ssh`, you should select the `tcp` subcommand. Using the `ssh` command is reserved for [credential injection](https://developer.hashicorp.com/boundary/tutorials/hcp-administration/hcp-ssh-cred-injection)

Reviewer 2: I believe the only valid target types are `tcp` and `ssh`, not `static`.

Also looking at the other domain model docs pages, we don't tend to include a `type` attribute in the list here. Instead the subtypes are listed below under sub-headings. Perhaps that would be a better place to call out that SSH Targets are required for credential injection, that you can only use SSH Targets if you want to use credential injection?

Author: Hey @tmessi , thanks for the feedback. Apologies on the 'static' reference. I could have sworn that I read that somewhere but probably got confused. I agree, perhaps under the SSH target attributes HCP/ENT section we could change the wording to:

`"SSH targets can source username/password or SSH private key credentials from Vault [credential libraries](https://developer.hashicorp.com/boundary/docs/concepts/domain-model/credential-libraries) or static [credentials](https://developer.hashicorp.com/boundary/docs/concepts/domain-model/credentials). If using injected crednetials, Boundary then injects the credentials into the SSH session between a client and end host. This allows users to securely connect to remote hosts using SSH, while never being in possession of a valid credential for that target host.

If the target type is set to SSH then brokered or injected credentials has to be used. You cannot reference a static .pem file to use`

Or something like that @Dan-Heath

Reviewer 2: It might help to state clearly the requirements around SSH Targets and injected application credentials:

- An SSH Target **must** have at least one injected application credential.
- A TCP Target **can not** have any injected application credentials.

Therefore:
- If you want to use brokered credentials when connecting to a target running SSH, you must use a TCP Target.
- If you want to use injected application credentials when connecting to a target running SSH, you must use an SSH Target.
- Also if you want to enable session recording for a target running SSH, you must use injected application credentials and an SSH Target.

### SSH target attributes <sup>HCP/ENT</sup>

SSH targets can source username/password or SSH private key credentials from Vault [credential libraries][] or static

> [credentials][]. If you use injected credentials, Boundary injects credentials into the SSH session between a client and end host. Injected credentials allow users to

Reviewer 2: Both the new wording and the previous wording seems a little confusing to me. We might want to make it clear again here that injected application credentials are required for an SSH target. Perhaps something like:

```
SSH targets use injected application credentials to authenticate
an SSH session between the client and end host.
This allows users to securely connect to remote hosts using SSH,
while never being in the possession of a valid credential for that target host.
The injected credentials can be
a username/password or username/private key credential
from Vault [credential libraries][] or static [credentials][],
or a SSH certificate from Vault [SSH certificate credential libraries][].
```

securely connect to remote hosts using SSH, while never being in possession of a valid credential for that target host.

<Note>

If you set the target `type` to `ssh`, you must use brokered or injected credentials.
You cannot reference a static .pem file.

</Note>

> 

Reviewer 2: This note is still confusing to me. Again an SSH target requires an injected application credential. You can also have brokered credentials in addition to the injected ones, but it is not a "borkered or injected".

Author: Yes this isn't right based on our conversation. Something like Tim put previously with a slight edit:

<Note>

If you want to use brokered credentials or a static .pem file when connecting to a target running SSH, you must set the target type to TCP

If you want to use injected application credentials when connecting to a target running SSH, you must set the target type to SSH

 </Note>

  Designates the storage bucket to be used for session recording.
  This attribute is required if you set `enable_session_recording` to `true`.

>   If using injected SSH credentials into a Boundary session, the `type` must be set to `ssh` and not `tcp`.

Reviewer: ```suggestion
  If you inject SSH credentials into a Boundary session, you must set the `type` to `ssh` and not `tcp`.
```
